### PR TITLE
[SPARK-7583][MLlib]User guide update for RegexTokenizer

### PR DESCRIPTION
**Pull request by hhbyyh**
jira: https://issues.apache.org/jira/browse/SPARK-7583

User guide update for RegexTokenizer

### CODE REVIEW

File: docs/ml-features.md [hunk 1]
```diff
@@ -217,21 +217,27 @@ for feature in result.select("result").take(3):
...
   (0, "Hi I heard about Spark"),
-  (0, "I wish Java could use case classes"),
-  (1, "Logistic regression models are neat")
+  (1, "I wish Java could use case classes"),
+  (2, "Logistic,regression,models,are,neat")
 )).toDF("label", "sentence")
 val tokenizer = new Tokenizer().setInputCol("sentence").setOutputCol("words")
-val wordsDataFrame = tokenizer.transform(sentenceDataFrame)
-wordsDataFrame.select("words", "label").take(3).foreach(println)
+val regexTokenizer = new RegexTokenizer().setInputCol("sentence").setOutputCol("words")
```
Comment (mengxr) [inline docs/ml-features.md]: Chop down setters:

~~~scala
val regexTokenizer = new RegexTokenizer()
  .setInputCol("sentence")
  .setOutputCol("words")
  .setPattern("\\W")
~~~

File: docs/ml-features.md [hunk 2]
```diff
@@ -217,21 +217,27 @@ for feature in result.select("result").take(3):
...
+  .setPattern("\\W")
+  // alternatively .setPattern("\\w+").setGaps(false)
```
Comment (mengxr) [inline docs/ml-features.md]: merge this line to the one above

File: docs/ml-features.md [hunk 3]
```diff
@@ -267,22 +274,28 @@ for (Row r : wordsDataFrame.select("words", "label").take(3)) {
   for (String word : words) System.out.print(word + " ");
   System.out.println();
 }
+
+RegexTokenizer regexTokenizer = new RegexTokenizer().setInputCol("sentence").setOutputCol("words")
```
Comment (mengxr) [inline docs/ml-features.md]: ditto. Chop down setters.

File: docs/ml-features.md [hunk 4]
```diff
@@ -267,22 +274,28 @@ for (Row r : wordsDataFrame.select("words", "label").take(3)) {
...
-  (1, "Logistic regression models are neat")
+  (1, "I wish Java could use case classes"),
+  (2, "Logistic regression models are neat")
 ], ["label", "sentence"])
 tokenizer = Tokenizer(inputCol="sentence", outputCol="words")
 wordsDataFrame = tokenizer.transform(sentenceDataFrame)
 for words_label in wordsDataFrame.select("words", "label").take(3):
   print(words_label)
+regexTokenizer = RegexTokenizer(inputCol="sentence", outputCol="words", pattern="\\W")
+# alternatively, pattern="\\w+", gaps(false)
```
Comment (mengxr) [inline docs/ml-features.md]: `gaps=False` (Python uses `False` instead of `false`.)

File: docs/ml-features.md [hunk 5]
```diff
@@ -217,21 +217,27 @@ for feature in result.select("result").take(3):
 
 [Tokenization](http://en.wikipedia.org/wiki/Lexical_analysis#Tokenization) is the process of taking text (such as a sentence) and breaking it into individual terms (usually words).  A simple [Tokenizer](api/scala/index.html#org.apache.spark.ml.feature.Tokenizer) class provides this functionality.  The example below shows how to split sentences into sequences of words.
 
-Note: A more advanced tokenizer is provided via [RegexTokenizer](api/scala/index.html#org.apache.spark.ml.feature.RegexTokenizer).
+[RegexTokenizer](api/scala/index.html#org.apache.spark.ml.feature.RegexTokenizer) allows more advanced tokenization based on regular expression (regex) matching. By default, the parameter pattern (regex, default: \\s+) is used as delimiters to split the input text. Alternatively, users can set parameter gaps to false indicating the regex pattern denotes "tokens" rather than splitting "gaps", and find all matching occurrences as the tokenization result.
```
Comment (mengxr) [inline docs/ml-features.md]: minor: try to limit the line width to 100 and make new line at the end of each sentence. This helps reduce the diff size when new changes come in.

Quote `pattern` and `gaps` so users know that they are parameter names.

File: docs/ml-features.md [hunk 6]
```diff
@@ -252,8 +259,8 @@ import org.apache.spark.sql.types.StructType;
 
 JavaRDD<Row> jrdd = jsc.parallelize(Lists.newArrayList(
   RowFactory.create(0, "Hi I heard about Spark"),
-  RowFactory.create(0, "I wish Java could use case classes"),
-  RowFactory.create(1, "Logistic regression models are neat")
+  RowFactory.create(1, "I wish Java could use case classes"),
+  RowFactory.create(2, "Logistic regression models are neat")
```
Comment (mengxr) [inline docs/ml-features.md]: Keep the example data in sync by replacing ` ` with `,` in this record.

File: docs/ml-features.md [hunk 7]
```diff
@@ -267,22 +274,28 @@ for (Row r : wordsDataFrame.select("words", "label").take(3)) {
...
 {% highlight python %}
-from pyspark.ml.feature import Tokenizer
+from pyspark.ml.feature import Tokenizer, RegexTokenizer
 
 sentenceDataFrame = sqlContext.createDataFrame([
   (0, "Hi I heard about Spark"),
-  (0, "I wish Java could use case classes"),
-  (1, "Logistic regression models are neat")
+  (1, "I wish Java could use case classes"),
+  (2, "Logistic regression models are neat")
```
Comment (mengxr) [inline docs/ml-features.md]: Keep the example data in sync by replacing ` ` with `,` in this record.

**Comment by hhbyyh** [comment]
Thanks for the review. I've sent an update addressing the comments.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.5. Thanks!

**Pull request closed without merging**